mail/views_test.py

```diff
@@ -38,6 +41,24 @@ def json(*args, **kwargs):  # pylint:disable=unused-argument, missing-docstring
     return json
 
 
+def mocked_batch_result(status_codes=None):
+    """Mocked version of the send_batch return value"""
+    if status_codes is None:
+        status_codes = [status.HTTP_200_OK]
+
+    def _is_response(_status_code):
+        """Helper function to figure out whether to generate a response or an exception"""
+        return isinstance(_status_code, int)
+    return [
+        (
+            ['recipient@example.com'],
+            Mock(spec=Response, status_code=status_code, json=mocked_json())
+            if _is_response(status_code) else None,
+            None if _is_response(status_code) else HTTPError(),
+        ) for status_code in status_codes
```

Comment:
There is _way_ too much logic going on here, and I can't follow it. What are you trying to achieve with this function? Can you make the code clearer? Readability counts.